extensions/schema-compare/src/dialogs/schemaCompareDialog.ts

```diff
@@ -376,7 +384,16 @@ export class SchemaCompareDialog {
 			return undefined;
 		}
 
+		let count = -1;
+		let idx = -1;
 		let values = cons.map(c => {
+			count++;
+
+			if (c.connectionId === this.connectionId) {
```

Comment:
should we check this.connectionId is present before this check? or wouldn't matter?

Reply:
it should be ok if this.connectionId is null/undefined since it just won't find a match here